raster-foundry/granary
# Refactor server startup and application config

## Overview

This PR:

- starts the server with `bloop` in `scripts/server`
- moves all of the config to command line + env variables + defaults (except for auth -- you have to say whether there's auth)
- runs both scalafix _and_ scalafmt in scripts/test
- adds sbt-explicit-dependencies to tests (is that the `etc.` in the issue? I hope so)

It does not make add a migrations command. It also doesn't do any work to make tests `bloop`-runnable in dev. It also doesn't remove the `api` service from docker-compose. It _also_ didn't move the tracing config to those options -- I wanted to leave it alone until we fix it, to avoid giving the impression that work had been done to it to whoever ends up trying to fix it.

I'm at about 3/4 time on the issue, so what else to do I think is a question of priority and convenience. The checklist from the issue is done (assuming I'm interpreting the etc. correctly) but I'm not sure if we want to do any of the things I haven't done so far.

Finally, it also assumes that you want auth on. Since we're going to default to auth on in general, I thought it made sense not to worry about how to disable auth conveniently.

### Checklist

none of the checklist items were relevant

## Testing Instructions

- `./scripts/server` without starting the database -- it should fail and show you a truly insane set of command line options and environment variables
- `docker-compose up -d database`, then `./scripts/server` again -- your server should start, and if you send a token from the `tokens` table in a JWT, you should be able to make some requests

Closes #50
